### PR TITLE
For #3700 - Add Setting to Delete Data on "Quit" menu action

### PR DESCRIPTION
**Pull request by ekager**
### Pull Request checklist

- [X] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [X] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [X] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [X] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features
Screenshot_20190910-171217
Screenshot_20190910-171221
Screenshot_20190910-171237

### CODE REVIEW

**Comment by ekager** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) **What questions will you answer with this data?**
- How often users tap "quit" from the browser toolbar menu

2) **Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?**

* To measure how often users interact with the quit button in Fenix to measure feature success and visibility.

3) **What alternative methods did you consider to answer these questions? Why were they not sufficient?**

* N/A (These are baseline metrics)

4) **Can current instrumentation answer these questions?**
* Currently there are no events to track this button

5) **List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data c](https://wiki.mozilla.org/Firefox/Data_Collection)[ollection ](https://wiki.mozilla.org/Firefox/Data_Collection)[categories](https://wiki.mozilla.org/Firefox/Data_Collection) on the found on the Mozilla wiki.**

* All data is Category 2.

6) **How long will this data be collected?**

Until 03/01/2020

7) **What populations will you measure?**

* All release, beta, and nightly users with telemetry enabled.

8) **Please provide a general description of how you will analyze this data.**

* Glean / Amplitude

9) **Where do you intend to share the results of your analysis?**

* Only on Glean, Amplitude and with mobile teams.

**Comment by ekager** [comment]
Tagging @boek for data review and @colintheshots for the feature since it was similar to granular data options and I ended up using your categories and had to modify some of the functions to work with mine

File: app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt [hunk 1]
```diff
@@ -252,7 +252,7 @@ sealed class Event {
         enum class Item {
             SETTINGS, LIBRARY, HELP, DESKTOP_VIEW_ON, DESKTOP_VIEW_OFF, FIND_IN_PAGE, NEW_TAB,
             NEW_PRIVATE_TAB, SHARE, REPORT_SITE_ISSUE, BACK, FORWARD, RELOAD, STOP, OPEN_IN_FENIX,
-            SAVE_TO_COLLECTION, ADD_TO_HOMESCREEN
+            SAVE_TO_COLLECTION, ADD_TO_HOMESCREEN, QUIT
```
Comment (liuche) [inline app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt]: @boek how do new events get into the metrics.md file? Do you have to change more files, or does the documentation automatically get added when you just add an event?

Should these be updated in the metrics.md file as part of the PR? At a quick glance I also didn't see save_to_collection or add_to_homescreen, so I'm wondering if I'm looking in the wrong place.

Comment (sblatz) [reply app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt]: I think we need to manually add this to `metrics.md` for now. At least until the Glean auto-generate documentation PR is merged 

@ekager

Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt]: Added to `metrics.md` and `metrics.yaml`!

Comment (liuche) [reply app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt]: Unrelated, @boek does this mean open_in_fenix, save_to_collection, and add_to_homescreen did not get data review, or need to be updated?

Comment (boek) [reply app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt]: @liuche that is correct
https://github.com/mozilla-mobile/fenix/pull/2131/files
and https://github.com/mozilla-mobile/fenix/pull/4925 didn't get data reviews. In the short term I could put myself as a codeowner of `metrics.kt` until we get a better process

File: app/metrics.yaml [hunk 2]
```diff
@@ -78,7 +78,7 @@ events:
         description: >
           A string containing the name of the item the user tapped. These items include:
           Settings, Library, Help, Desktop Site toggle on/off, Find in Page, New Tab,
-          Private Tab, Share, Report Site Issue, Back/Forward button, Reload Button
+          Private Tab, Share, Report Site Issue, Back/Forward button, Reload Button, Quit
     bugs:
     - 1024
     data_reviews:
```
Comment (boek) [inline app/metrics.yaml]: Could we add a link to this PR for this data review?

Comment (ekager) [reply app/metrics.yaml]: Just did!

**Comment by boek** [comment]
1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?

Yes, updating existing probes w/ documentation

2) Is there a control mechanism that allows the user to turn the data collection on and off?

Yes, Fenix data controls

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

Fenix PM Vesta will monitor, and this has an expiry

4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Type 2, interaction with search UI, Type 1 for search w/ default search engine (bundled, not custom)

5) Is the data collection request for default-on or default-off?

Default on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does there need to be a check-in in the future to determine whether to renew the data? (Yes/No) (If yes, set a todo reminder or file a bug if appropriate)**

Has expiry, 3/2020

9) Does the data collection use a third-party collection tool? **If yes, escalate to legal.**

No

File: app/src/main/java/org/mozilla/fenix/settings/DeleteBrowsingDataOnQuitFragment.kt [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        setPreferencesFromResource(R.xml.delete_browsing_data_quit_preferences, rootKey)
+    }
+
+    @Suppress("ComplexMethod")
+    override fun onResume() {
+        super.onResume()
+        activity?.title = getString(R.string.preferences_delete_browsing_data_on_quit)
+        (activity as AppCompatActivity).supportActionBar?.show()
+
+        class CheckboxUpdater : SharedPreferenceUpdater() {
```
Comment (boek) [inline app/src/main/java/org/mozilla/fenix/settings/DeleteBrowsingDataOnQuitFragment.kt]: Creating an actual class inside of this method doesn't look right. Could we just create the updater inline

```
val checkboxUpdater = object : SharedPreferenceUpdater() {
...
}
```
and then use that as the listener?

Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/settings/DeleteBrowsingDataOnQuitFragment.kt]: wow yep i knew this looked weird lol thanks!

**Pull request merged**